arm-image-installer

```diff
@@ -606,6 +632,34 @@ if [ "$SSH_KEY" != "" ]; then
 		echo "= WARNING: No SSH Key Added."
 	fi
 fi
+# Configure Wi-Fi if SSID and password are provided
+if [ "$WIFI_SSID" != "" ] && [ "$WIFI_PASS" != "" ]; then
+    echo "= Configuring Wi-Fi: $WIFI_SSID"
+    mkdir -p ${PREFIX}/etc/NetworkManager/system-connections/
+    # Use the SSID as the filename (replace spaces and special characters with underscores)
+    FILENAME=$(echo "$WIFI_SSID" | tr ' ' '_' | tr -cd '[:alnum:]._-').nmconnection
+    cat > ${PREFIX}/etc/NetworkManager/system-connections/$FILENAME <<-EOF
+[connection]
+id=$WIFI_SSID
+type=wifi
+interface-name=wlan0
+
+[wifi]
+mode=infrastructure
+ssid=$WIFI_SSID
+
+[wifi-security]
+key-mgmt=wpa-psk
+psk=$WIFI_PASS
+
+[ipv4]
+method=auto
+
+[ipv6]
+method=auto
```

Comment:
Looking through my configs this usually, but not always has addr-gen-mode=default

Reply:
added....